VitDeck/Assets/VitDeck/Validator/Rules/Booth/B01_BoothBoundsRule.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+namespace VitDeck.Validator
+{
+    /// <summary>
+    /// ブースのサイズ制限を調べるルール
+    /// </summary>
+    public class BoothBoundsRule : BaseRule
+    {
+        private readonly Bounds limit;
+
+        /// <summary>
+        /// コンストラクタ。
+        /// </summary>
+        /// <param name="name">ルールの名前</param>
+        /// <param name="size">バウンディングボックスの大きさ</param>
+        /// <param name="margin">制限に持たせる余裕</param>
+        public BoothBoundsRule(string name, Vector3 size, float margin)
+            : this(name, size, margin, pivot: Vector3.zero) { }
+
+        /// <summary>
+        /// コンストラクタ。
+        /// </summary>
+        /// <param name="name">ルールの名前</param>
+        /// <param name="size">バウンディングボックスの大きさ</param>
+        /// <param name="margin">制限に持たせる余裕</param>
+        /// <param name="pivot">バウンディングボックスの原点（中心下）</param>
+        public BoothBoundsRule(string name, Vector3 size, float margin, Vector3 pivot) : base(name)
+        {
+            var center = pivot + (Vector3.up * size.y * 0.5f);
+            var limit = new Bounds(center, size);
+            limit.Expand(margin);
+            this.limit = limit;
+        }
+
+        protected override void Logic(ValidationTarget target)
+        {
+            var oversizes = target
+                .GetAllObjects()
+                .Where(gameObject => gameObject.activeSelf && gameObject.activeInHierarchy)
+                .SelectMany(GetObjectBounds)
+                .Where(data => !LimitContains(data.bounds));
+
+            foreach (var oversize in oversizes)
+            {
+                AddIssue(new Issue(oversize.objectReference, IssueLevel.Error, string.Format("オブジェクトがブースサイズ制限を超えています。（制限={0}, オブジェクト={1}, オブジェクトの種類={2}）", limit, oversize.bounds, oversize.objectReference.GetType().Name)));
```

Comment:
細かいですが、見た目整形のみ指摘です。
```suggestion
                AddIssue(new Issue(oversize.objectReference, IssueLevel.Error, string.Format("オブジェクトがブースサイズ制限を超えています。{3}制限={0}{3}対象={1}{3}オブジェクトの種類={2}", limit, oversize.bounds, oversize.objectReference.GetType().Name, System.Environment.NewLine)));
```


![image](https://user-images.githubusercontent.com/46106857/59565400-404c2a80-908e-11e9-8d9c-b4fb1396114e.png)